generator/plugins/proto2gql/messages_inputs.go

```diff
@@ -56,71 +56,118 @@ func (g *Proto2GraphQL) inputObjectMapFieldTypeResolver(mapFile *parsedFile, mp
 func (g *Proto2GraphQL) fileInputObjects(file *parsedFile) ([]graphql.InputObject, error) {
 	var res []graphql.InputObject
 	for _, msg := range file.File.Messages {
-		if !msg.HaveFields() {
+		fields, err := g.getMessageFields(file, msg)
+		if err != nil {
+			return nil, err
+		}
+
+		if fields == nil {
 			continue
 		}
-		var fields []graphql.ObjectField
-		for _, field := range msg.Fields {
-			fieldTypeFile, err := g.parsedFile(field.Type.File())
-			if err != nil {
-				return nil, errors.Wrap(err, "failed to resolve value type file")
-			}
-			typ, err := g.inputMessageFieldTypeResolver(fieldTypeFile, field)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to resolve `%s.%s` field type", msg.Name, field.Name)
-			}
 
-			msgCfg, err := file.Config.MessageConfig(msg.Name)
+		// TODO: oneof fields
+		res = append(res, graphql.InputObject{
+			VariableName: g.inputMessageVariable(file, msg),
+			GraphQLName:  g.inputMessageGraphQLName(file, msg),
+			Fields:       fields,
+		})
+	}
+
+	return res, nil
+}
+
+func (g *Proto2GraphQL) getMessageFields(file *parsedFile, msg *parser.Message) ([]graphql.ObjectField, error) {
+	if !msg.HaveFields() {
+		return nil, nil
+	}
+
+	msgCfg, err := file.Config.MessageConfig(msg.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve message %s config", msg.Name)
+	}
+
+	var fields []graphql.ObjectField
+	for _, field := range msg.Fields {
+		fldCfg := msgCfg.Fields[field.Name]
+		if fldCfg.ContextKey != "" {
+			continue
+		}
+
+		fieldTypeFile, err := g.parsedFile(field.Type.File())
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to resolve value type file")
+		}
+
+		fieldTypeMessage, ok := field.Type.(*parser.Message)
+
+		if ok {
+			fieldTypeMsgCfg, err := fieldTypeFile.Config.MessageConfig(fieldTypeMessage.Name)
 			if err != nil {
-				return nil, errors.Wrapf(err, "failed to resolve message %s config", msg.Name)
+				return nil, errors.Wrapf(err, "failed to resolve message %s config", fieldTypeMessage.Name)
 			}
-			fldCfg := msgCfg.Fields[field.Name]
-			if fldCfg.ContextKey != "" {
+
+			if fieldTypeMsgCfg.UnwrapField {
```

Comment:
maybe we should move logic of this if block to separate function?

Reply:
Done, PTAL.